### PR TITLE
Require configuration for tracing.

### PR DESCRIPTION
**Pull request by olix0r**
Problem

Falling back to use DefaultTracer causes log noise for most configurations.

Solution

Do not fall back to DefaultTracer. Instead, if no telemeter is configured, we
provide a default telemeter that has tracing disabled.

### CODE REVIEW

File: linkerd/core/src/main/scala/io/buoyant/linkerd/Linker.scala [hunk 1]
```diff
@@ -88,44 +88,48 @@ object Linker {
...
-            DefaultStatsReceiver
-          case receivers =>
-            for (r <- receivers) log.info(s"Using stats receiver: $r")
-            BroadcastStatsReceiver(receivers)
+      val stats = {
+        val receivers = telemeters.collect { case t if !t.stats.isNull => t.stats } match {
+          case Nil => NullStatsReceiver
+          case receivers => BroadcastStatsReceiver(receivers)
         }
+        for (t <- tracers) log.info("stats: %s", t)
```
Comment (adleong) [inline linkerd/core/src/main/scala/io/buoyant/linkerd/Linker.scala]: should this be `r <- receivers`?

File: linkerd/core/src/main/scala/io/buoyant/linkerd/Linker.scala [hunk 2]
```diff
@@ -88,44 +88,48 @@ object Linker {
...
-      val stats =
-        telemeters.getOrElse(Nil).collect { case t if !t.stats.isNull => t.stats } match {
-          case Nil =>
-            log.info(s"Using default stats receiver")
-            DefaultStatsReceiver
-          case receivers =>
-            for (r <- receivers) log.info(s"Using stats receiver: $r")
-            BroadcastStatsReceiver(receivers)
+      val stats = {
+        val receivers = telemeters.collect { case t if !t.stats.isNull => t.stats } match {
```
Comment (adleong) [inline linkerd/core/src/main/scala/io/buoyant/linkerd/Linker.scala]: should this name be singular instead of plural?

**Comment by adleong** [comment]
Let me see if I've got this straight:

# Old

* No telemeters => Default tracer + default stats receiver
* Some telemeters but none that provide stats => default stats receiver
* Some telemeters but none that provide tracers => default tracer

# New 

* No telemeters => Null tracer + default stats receiver
* Some telemeters but none that provide stats => default stats receiver
* Some telemeters but none that provide tracers => null tracer

What even is the default tracer?  Is it ever useful to us?

**Comment by olix0r** [comment]
The default tracer is "whatever was loaded on the classpath".  At twitter, this is finagle-zipkin, configured to write to localhost:SCRIBEPORT.  I don't think linkerd should maintain this behavior.

You're right with the one exception that the new behavior is:
- Some telemeters but none that provide stats => null stats receiver

File: linkerd/core/src/main/scala/io/buoyant/linkerd/Linker.scala (hunk 1 above)
Comment (olix0r) [inline linkerd/core/src/main/scala/io/buoyant/linkerd/Linker.scala]: ah yeah, i thought i fixed this (probably in another branch)

File: linkerd/core/src/main/scala/io/buoyant/linkerd/Linker.scala (hunk 2 above)
Comment (olix0r) [inline linkerd/core/src/main/scala/io/buoyant/linkerd/Linker.scala]: correct

**Comment by adleong** [comment]
:star: 👡

**Pull request merged**